lib/github_changelog_generator/octo_fetcher.rb

```diff
@@ -280,19 +281,47 @@ def iterate_pages(client, method, *args)
     #
     # @return [Object] returns exactly the same, what you put in the block, but wrap it with begin-rescue block
     def check_github_response
+      attempt = 1
       begin
         value = yield
       rescue Octokit::Unauthorized => e
         Helper.log.error e.message
-        abort "Error: wrong GitHub token"
+        sys_abort("Error: wrong GitHub token")
       rescue Octokit::Forbidden => e
+        attempt += 1
+        sleep_time = exp_backoff(attempt)
+        Helper.log.warn("sleeping #{sleep_time} second")
+        sys_sleep(sleep_time)
+
         Helper.log.warn e.message
         Helper.log.warn GH_RATE_LIMIT_EXCEEDED_MSG
         Helper.log.warn @client.rate_limit
+
+        if attempt >= MAX_FORBIDDEN_RETRIES
+          sys_abort("Exceeded retry limit")
+        else
+          retry
+        end
       end
       value
     end
 
+    def sys_sleep(seconds)
+      sleep(seconds)
+    end
+
+    def sys_abort(msg)
+      abort(msg)
+    end
+
+    # Returns the exponential backoff (seconds) for this attempt number
+    #
+    # @param [Integer] attempt the attempt number
+    # @return [Integer] Exponential backoff seconds
+    def exp_backoff(attempt)
+      (2**attempt - 1) / 2
```

Comment:
You're right. It's not very explicit in their API docs. It's actually under the API limit, but Github flags certain requests that match their "abuse detection". It happens frequently for me (large git repo) when fetching tag-dates. I believe this is a simple API call, that happens very rapidly, 100s of times in a row. Github sees this as potentially abuse, so they give the `Octokit::Forbidden` error.